sui/src/unit_tests/cli_tests.rs

```diff
@@ -1362,8 +1362,12 @@ async fn test_split_coin() -> Result<(), anyhow::Error> {
 
     let object_refs = context.gateway.get_owned_objects(address)?;
 
-    // Check log output contains all object ids.
-    let coin = object_refs.get(1).unwrap().0;
+    // Get another coin
```

Comment:
Not related to this PR, but I believe you no longer need to call `sync` after a transaction. The only time we need sync should be for genesis objects.

Reply:
With https://github.com/MystenLabs/sui/pull/1042, we were able to store every new object in the effects into the gateway storage.

Reply:
Xun have refactored the gateway to use GatewayStore instead of map of AccountState a while back, you don't need to call sync if the affected accounts are managed in the same gateway, the tx effects are stored to a unified GatewayStore.